src/badge_manager_standalone.py

```diff
@@ -36,22 +36,42 @@ def _read_file(self,device_file):
         self.logger.info("Reading devices from file: {}".format(device_file))
 
         regex = re.compile(r'^([A-Fa-f0-9]{2}(?::[A-Fa-f0-9]{2}){5}).*')
+       
+        #extracting badge id and project from device file
+        with open(device_file, 'r') as devices_macs:
```

Comment:
the whole way the file if being read right now is very messy, please clean up the code.
* Change the order of fields to keep the badge name at the end (since it may contain spaces, and we don't use it here anyway yet). Should be mac badge_id project_id name
* The original code validates the structure of the line, and accepts only lines that start with a MAC address (that's what this funny regular expression does). If a line is commented (with #), it will ignore it.
* You are currently reading the file twice. This leaves room for potential issues. For example, if the file has 3 lines, one of which is commented. Your code will read 3 lines, but the code that reads the mac addresses will only read 2 lines. This will cause issues further down when you combine the information
* The right way to do this, would be to fix the regex code so it extracts the badge_id and project id as well. Then, read each line (mac, badge_id, project_id) into a dict object, and later use this dict object to create the badges).
* If the current code is confusing  (there's some python magic that your predecessor used...), feel free to simplify and and use easier to read code.

Reply:
- Changed the order of fields.Now, it is mac, badge id, project id , name
-Now I am reading the file only once.Now, I am extracting mac address, badge id and project id without using regex.Let me know if you want me to do specifically using regex due to any issue but the way I am doing is quite simple.Just ignoring the lines starting with "#"
- Please ignore changes in file hub_manager.py, nrf.py, sever.py & settings.py .It's just spaces.I have already committed without checking it.Next time, I will keep in mind to use git diff.My apology for the inconvenience.

Reply:
I'm ok with the simplified version. It's not a critical piece of code, so simple (and readable) is good.